src/proof/proof_rule.h

```diff
@@ -1385,241 +1385,349 @@ enum class PfRule : uint32_t
    * \endverbatim
    */
   QUANTIFIERS_PREPROCESS,
-  //================================================= String rules
-  //======================== Core solver
-  // ======== Concat eq
-  // Children: (P1:(= (str.++ t1 ... tn t) (str.++ t1 ... tn s)))
-  // Arguments: (b), indicating if reverse direction
-  // ---------------------
-  // Conclusion: (= t s)
-  //
-  // Notice that t or s may be empty, in which case they are implicit in the
-  // concatenation above. For example, if
-  // P1 concludes (= x (str.++ x z)), then
-  // (CONCAT_EQ P1 :args false) concludes (= "" z)
-  //
-  // Also note that constants are split, such that if
-  // P1 concludes (= (str.++ "abc" x) (str.++ "a" y)), then
-  // (CONCAT_EQ P1 :args false) concludes (= (str.++ "bc" x) y)
-  // This splitting is done only for constants such that Word::splitConstant
-  // returns non-null.
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Stings -- Core rules -- Concatenation equality**
+   *
+   * .. math::
+   *
+   *   \inferrule{(t_1\cdot\ldots \cdot t_n \cdot t) = (t_1 \cdot\ldots
```

Comment:
I would prefer to denote concatenation by `\circ` instead of `\cdot`.

Reply:
I was indeed following the notation used in cvc5 string papers. What do you think @ajreynol ?

Reply:
\cdot is used for concatentation.